aim/ext/task_queue/queue.py

```diff
@@ -0,0 +1,62 @@
+import threading
+import queue
+import atexit
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class TaskQueue(object):
+    def __init__(self, name, num_workers=1, max_backlog=0):
+        self.name = name
+        self.num_workers = num_workers
+
+        self._queue = queue.Queue(maxsize=max_backlog)
+
+        self._threads = []
+        self._shutdown = False
+        self._stopped = False
+        atexit.register(self.stop_workers)
+
+        for thread_num in range(self.num_workers):
+            thread = threading.Thread(target=self.worker)
+            thread.daemon = True
+            self._threads.append(thread)
+            thread.start()
+
+    def register_task(self, task_func, *args, **kwargs):
+        if self._stopped:
+            logger.debug('Cannot register task: task queue is stopped.')
+        else:
+            self._queue.put((task_func, args, kwargs))
```

Comment:
Let's do two things here:
1. Having a nonzero (though a sufficiently big number) `max_backlog`
2. Warn a user something like:
     > It seems like you are logging too often which may cause performance degradation. Consider logging / tracking every 100 iterations instead.

(1) is not mandatory as we can log such a warning even before `Queue.maxsize` is reached.

Reply:
Re: max queue size. I was keeping this in mind; just need to know what we consider as a reasonabe number. Any suggestions?
Re: warning message. How can we know user tracks once every N iterations? Maybe something more generic, like `'Consider tracking at lower pace'`

Reply:
IDK, maybe 1-10 million? Maybe less, but not more (10 million of plain float writes is already > 100MB, let's not use too much memory, we better block the main thread and log a warning).

Yes, `'Consider tracking at a lower pace'` is much better. Also, what if we have FAQ section in our docs and additionally refer to them (output URLs) for this and other such cases.

Reply:
Yeah, having FAQ section is a great idea. Will add TODO item for that. For now lets use 10mln. as a max size and warn user when queue is 80% full.